java-checks/src/main/java/org/sonar/java/checks/MathClampRangeCheck.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.java.annotations.VisibleForTesting;
+import org.sonar.java.checks.helpers.QuickFixHelper;
+import org.sonar.java.checks.methods.AbstractMethodDetection;
+import org.sonar.java.reporting.JavaQuickFix;
+import org.sonar.java.reporting.JavaTextEdit;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.JavaVersion;
+import org.sonar.plugins.java.api.JavaVersionAwareVisitor;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+
+import static org.sonar.java.model.SyntacticEquivalence.areEquivalentIncludingSameVariables;
+
+@Rule(key = "S6913")
+public class MathClampRangeCheck extends AbstractMethodDetection implements JavaVersionAwareVisitor {
+
+  public static final String DOUBLE = "double";
+  public static final String FLOAT = "float";
+  public static final String INT = "int";
+  public static final String LONG = "long";
+
+  public static final String MIN = "min";
+  public static final String MAX = "max";
+  public static final String VALUE = "value";
+
+  @Override
+  public boolean isCompatibleWithJavaVersion(JavaVersion version) {
+    return version.isJava21Compatible();
+  }
+
+  @Override
+  protected MethodMatchers getMethodInvocationMatchers() {
+    return MethodMatchers.create()
+      .ofTypes("java.lang.Math")
+      .names("clamp")
+      .addParametersMatcher(DOUBLE, DOUBLE, DOUBLE)
+      .addParametersMatcher(FLOAT, FLOAT, FLOAT)
+      .addParametersMatcher(LONG, LONG, LONG)
+      .addParametersMatcher(LONG, INT, INT)
+      .build();
+  }
+
+  @Override
+  protected void onMethodInvocationFound(MethodInvocationTree mit) {
+    if (mit.arguments().size() == 3) {
```

Comment:
If this check is really necessary maybe let's keep track of why with a comment here. People revisiting this rule in the future might simply remove it otherwise